spec/exoscale/requests/security_groups_spec.rb

```diff
@@ -0,0 +1,107 @@
+      @client.list_security_groups['listsecuritygroupsresponse']['count'].must_equal @count_of_sg + 1
+    end
+    
+    it "must list the new security group" do
+      @client.list_security_groups['listsecuritygroupsresponse']['securitygroup'].select{|sg| sg['name']==@new_sg['name']}[0]['id'].must_equal @new_sg['id']
+    end
+    
+    after do
+      begin
+        @client.delete_security_group(:name => @new_sg['name'])
+      rescue Fog::Compute::Exoscale::Error => e
+        if e.message == "Cannot delete group when it's in use by virtual machines"
+          sleep 30
+          retry
+        else
+          raise Fog::Compute::Exoscale::Error, :message => e.message
+        end
+      end
+    end
+  end
+  
+  describe "when authorizing rules to a new security group" do
+    before do
+      @another_sg = @client.create_security_group(:name => "fog_exoscale_test_sg_rules_#{Random.new.rand(1..2000)}")['createsecuritygroupresponse']['securitygroup']
+      # empty security group
+    end
+    
+    it "must have no rules" do
+      @another_sg['ingressrule'].must_be_empty
+      @another_sg['egressrule'].must_be_empty
+    end
+    
+    describe "When adding new rules" do
+      before do
+        # adding an ingress rule:
+        #  keeping the job id
+        ingress_job_id = @client.authorize_security_group_ingress(:securityGroupName => @another_sg['name'], :cidrList => '0.0.0.0/0', :startPort => '22', :endPort => '22', :protocol => 'TCP')['authorizesecuritygroupingressresponse']['jobid']
+        sleep 2 # because it's an async request
+        #  then querying the async job  
+        ingress_result = @client.query_async_job_result(:jobId => ingress_job_id)['queryasyncjobresultresponse']
+        while (ingress_result['jobresult'] == nil)
+          sleep 2
+          ingress_result = @client.query_async_job_result(:jobId => ingress_job_id)['queryasyncjobresultresponse']
+        end
+        #  and finally retrieving the rule id so we can delete it later
+        @ingress_rule_id = ingress_result['jobresult']['securitygroup']['ingressrule'][0]['ruleid']
+        
+        # and same for egress rule
+        egress_job_id = @client.authorize_security_group_egress(:securityGroupName => @another_sg['name'], :cidrList => '0.0.0.0/0', :startPort => '22', :endPort => '22', :protocol => 'TCP')['authorizesecuritygroupegressresponse']['jobid']
+        sleep 2
+        egress_result = @client.query_async_job_result(:jobId => egress_job_id)['queryasyncjobresultresponse']
+        while (egress_result['jobresult'] == nil)
+          sleep 2
+          egress_result = @client.query_async_job_result(:jobId => egress_job_id)['queryasyncjobresultresponse']
+        end
+        @egress_rule_id = egress_result['jobresult']['securitygroup']['egressrule'][0]['ruleid']
+        
+        @another_sg = @client.list_security_groups['listsecuritygroupsresponse']['securitygroup'].select{|sg| sg['name']==@another_sg['name']}[0]
+      end
```

Comment:
Wow. What a monster setup. Can't we improve this?
It seems that the code itself does not help much and thus the test reflects it.

Not a blocker, just a comment.